rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -1883,439 +1857,228 @@ fn check_stream_handler_generated_reject_response_canister_migrating() {
-                    ],
-                    1,
-                ),
-                (
-                    &[
-                        (LABEL_TYPE, LABEL_VALUE_TYPE_REQUEST),
-                        (LABEL_STATUS, LABEL_VALUE_SENDER_SUBNET_MISMATCH),
-                    ],
-                    2,
-                ),
-                (
-                    &[
-                        (LABEL_TYPE, LABEL_VALUE_TYPE_RESPONSE),
-                        (LABEL_STATUS, LABEL_VALUE_SUCCESS),
-                    ],
-                    1,
-                ),
-                (
-                    &[
-                        (LABEL_TYPE, LABEL_VALUE_TYPE_RESPONSE),
-                        (LABEL_STATUS, LABEL_VALUE_CANISTER_NOT_FOUND),
-                    ],
-                    1,
-                ),
-            ]),
-            &metrics_registry,
-        );
-        assert_eq!(
-            3,
-            fetch_inducted_payload_sizes_stats(&metrics_registry).count
-        );
+    metrics.assert_inducted_xnet_messages_eq(&[
+        // Requests @43 and @44 successfully inducted.
+        (LABEL_VALUE_TYPE_REQUEST, LABEL_VALUE_SUCCESS, 2),
+        // Request @46 not inducted because of missing canister.
+        (LABEL_VALUE_TYPE_REQUEST, LABEL_VALUE_CANISTER_NOT_FOUND, 1),
+        // Request @47 not inducted because of canister not on `REMOTE_SUBNET`.
+        // Request @48 not inducted becaue of unknown canister sender.
+        (
+            LABEL_VALUE_TYPE_REQUEST,
+            LABEL_VALUE_SENDER_SUBNET_MISMATCH,
+            2,
+        ),
+        // Response @45 successfully inducted.
+        (LABEL_VALUE_TYPE_RESPONSE, LABEL_VALUE_SUCCESS, 1),
+        // Response @49 not inducted because of missing canister.
+        (LABEL_VALUE_TYPE_RESPONSE, LABEL_VALUE_CANISTER_NOT_FOUND, 1),
+    ]);
+    assert_eq!(3, metrics.fetch_inducted_payload_sizes_stats().count);
+    // Three critical errors raised.
+    metrics.assert_eq_critical_errors(CriticalErrorCounts {
+        induct_response_failed: 1,
+        sender_subnet_mismatch: 2,
+        ..CriticalErrorCounts::default()
     });
 }
 
-/// Tests that a response to a missing canister is dropped, incrementing the
-/// respective critical error count.
+/// Tests that a message addressed to a canister that is not currently hosted by
+/// this subnet; and is not being migrated on a path containing both this subnet
+/// and its known host; is dropped, incrementing the respective critical error
+/// count.
 #[test]
-fn induct_stream_slices_response_to_missing_canister() {
-    with_test_replica_logger(|log| {
-        let (stream_handler, mut initial_state, metrics_registry) = new_fixture(&log);
-
-        let mut expected_state = initial_state.clone();
-
-        // Initial state with no canisters and one stream.
-        let outgoing_stream = generate_outgoing_stream(StreamConfig {
-            messages_begin: 21,
-            message_count: 0,
```

Comment:
This test is redundant since it's functionality is completely encompassed by the more complex test `induct_stream_slices_partial_success()`. The only thing this test was missing was the check on the critical error counter, which should have been there anyway. Considering there are even more complicated tests in this file, I see no reason to check this in isolation.